### PR TITLE
Refactored ArtifactManager API

### PR DESCRIPTION
**Pull request by chhex**
The ArtifactManager API for listing the Dependencies in the DependencyManagement Section of a Bom has been changed to take the Maven Coordinates of the Bom as Parameter: 

 `	List<MavenArtifact> listDependenciesInBom(MavenArtifact bom);

	List<MavenArtifact> listDependenciesInBom(MavenArtifact bom, SearchCondition searchFilter);`

instead of being "hardcoded" in ArtifactManagerImpl

Also the Piper Service Methods:

```
/**
	 * @param serviceName Patch, for which Artifacts are listed
	 * @param filter Filter, which should be applied for search
	 * @return list of Maven Arifacts relevant for Patch
	 */
    List<MavenArtifact> listMavenArtifacts(String serviceName, SearchCondition filter);
	/**
	 * List all Maven Artifacts 
	 * @return list of Maven Arifacts
	 * @param serviceName the Name of the Service for which the Artifacts are listed
	 */
    List<MavenArtifact> listMavenArtifacts(String serviceName);
```

have been changed to retrieve the Bom Coordinates from the ServiceMetaData. 

```
	@Override
	public List<MavenArtifact> listMavenArtifacts(String serviceName, SearchCondition filter) {
		ServiceMetaData data = repo.getServiceMetaDataByName(serviceName);
		return am.listDependenciesInBom(data.getBomCoordinates(), filter);
	}

```
This way it should easier to accommodate the UI requirements in terms of Selection of what is to be patched. 

Also RuntimeException have be reversed to  PatchServiceRuntimeException

### CODE REVIEW

**Comment by chhex** [comment]
@apgsga-jhe You can merge, after your review , whenever suits you.

File: apg-patch-artifact-querymanager/src/main/java/com/apgsga/artifact/query/impl/ArtifactManagerImpl.java [hunk 1]
```diff
@@ -63,39 +60,8 @@
 
     private final RepositorySystemSession session;
 
-    private String bomGroupId = DEFAULT_BOM_GROUP_ID;
-
-    private String bomArtefactId = DEFAULT_BOM_ARTIFACT_ID;
-
     private final String localRepo;
```
Comment (apgsga-jhe) [inline apg-patch-artifact-querymanager/src/main/java/com/apgsga/artifact/query/impl/ArtifactManagerImpl.java:63]: @chhex : Is local Repo still used?

Comment (apgsga-jhe) [reply apg-patch-artifact-querymanager/src/main/java/com/apgsga/artifact/query/impl/ArtifactManagerImpl.java:63]: @chhex : Sorry, forget my comment ..... the file was actually collapse in Github, and didn't see the whole code :/ ...

**Review by apgsga-jhe — APPROVED** [review]
@chhex : this is looking good to me. I'll merge and try to integrate that ASAP on my dev branch so that it gets indirectly tested during my tests.

**Pull request merged**